app/utilities/metadata_parser.py

```diff
@@ -146,6 +146,27 @@ def convert_schema_name(self, data, **kwargs):
             )
         return data
 
+    @post_load
+    def metadata_response_id(
+        self, data, **kwargs
+    ):  # pylint: disable=no-self-use, unused-argument
+        """
+        If response_id is not present : Build response_id from ru_ref,collection_exercise_sid,eq_id and form_type
+                                        and updates metadata with response_id
+        If response_id is present : return as it is
+
+        """
+        if data.get("response_id"):
+            return data
+        ru_ref = data.get("ru_ref")
+        collection_exercise_sid = data.get("collection_exercise_sid")
+        eq_id = data.get("eq_id")
+        form_type = data.get("form_type")
+        if ru_ref and collection_exercise_sid and eq_id and form_type:
```

Comment:
This should raise a `ValidationError` for the `else` case.

Reply:
It is not needed `marshmallow` handles it as part of `validation`

Reply:
Although true, I don't think that is explicit and replies on the order and the fact that `schema_name` is optional. If schema_name became mandatory, then our test wouldn't fail, but we may not have eq_id/form_type.

Reply:
The else part doesn't get executed , it is caught before that in [here](https://github.com/ONSdigital/eq-questionnaire-runner/blob/master/app/utilities/metadata_parser.py#L131-L133)

Reply:
ignore , above one. i tweaked test to make sure `else` part `ValidationError` gets raised

Reply:
Yes, I understand that, but that solely dependents on the logic for `schema_name`, if the schema name existed but eq_id or form_type didn't, this wouldn't raise a validation error.